### PR TITLE
[Bot] Implement from issue #108: Create simple math utility with basic operations

### LINKED ISSUE #108: Create simple math utility with basic operations
﻿## Project Tag
project: math

## Description
Create a lightweight math utility module that provides basic arithmetic operations for use in other projects.

## Acceptance Criteria
- Function to add two numbers
- Function to multiply two numbers  
- Basic input validation
- Simple test to verify functions work
- Documentation with usage example

## Files to create
- \`math_utils.py\` - main utility functions
- \`test_math_utils.py\` - basic tests
- \`README.md\` - usage documentation

## Dependencies
- No external dependencies
- Use Python standard library only

## Scope & Segmentation Guidelines
**IMPORTANT FOR ANALYZER**: This is a simple, focused project. 
- Create maximum 3-4 logical tasks
- Group related activities together (don't create separate tasks for each function)
- Suggested segmentation: (1) Core implementation, (2) Testing, (3) Documentation
- Avoid micro-tasks like 'create __init__.py' or 'add single function'
- Keep tasks substantial and meaningful

This is a workflow verification test - prioritize logical task grouping over granular breakdown."

### PR DESCRIPTION
**Pull request by morotrade**
Auto-generated implementation.

Closes #108

Project root: `projects`

### CODE REVIEW

**Comment by morotrade** [comment]
### 🤖 AI Code Review



**Last Updated**: 2025-09-13 23:20:30 UTC

#### 📂 Project Root
`projects`

#### 📊 Summary
The implementation of the math utility is functional but requires improvements in testing practices and usability.

#### 🎯 Issue Counts
- **🚫 BLOCKER**: 0
- **⚠️ IMPORTANT**: 1  
- **💡 SUGGESTION**: 2

#### 🔍 Detailed Findings

#### ⚠️ IMPORTANT
- **`projects/math_utils.py:12`**: The module lacks a proper entry point for testing and usage.
  *Suggestion*: Consider adding a main function that allows for command-line usage of the utility.

#### 💡 SUGGESTION
- **`projects/test_math_utils.py:8`**: The test for the add function does not check for exceptions properly.
  *Suggestion*: Use 'with self.assertRaises(ValueError):' context manager for better readability.
- **`projects/test_math_utils.py`**: Consider adding more test cases for edge cases.
  *Suggestion*: Add tests for negative numbers, zero, and large numbers.


#### 🔧 Suggested Patches
The following unified diffs stay under the project root and can be applied by the developer:

**Patch 1**

```diff
--- a/projects/math_utils.py
+++ b/projects/math_utils.py
@@ -11,5 +11,8 @@
     return a * b
 
+def main():
+    print("This is a simple math utility. Use add(a, b) or multiply(a, b).")
+
 if __name__ == "__main__":
-    print("Math Utility Module")
+    main()
```

**Patch 2**

```diff
--- a/projects/test_math_utils.py
+++ b/projects/test_math_utils.py
@@ -7,7 +7,7 @@
         self.assertRaises(ValueError, add, 2, "three")
+        with self.assertRaises(ValueError):
+            add(2, "three")
 
     def test_multiply(self):
         self.assertEqual(multiply(2, 3), 6
```

---
> 🔄 **Auto-Review Loop**: This comment updates automatically when you push changes to this branch.  
> 🏷️ **Labels**: `need-fix` = blockers to resolve, `ready-to-merge` = all clear!

**Comment by morotrade** [comment]
🔎 Using project root: `projects` for PR-fix

**Comment by morotrade** [comment]
❌ Failed to apply LLM patch in PR-fix mode. See logs.

**Comment by morotrade** [comment]
🔎 Using project root: `projects` for PR-fix
